src/hooks/useLinearPrimaryTicks.ts

```diff
@@ -0,0 +1,98 @@
+import type { ScaleContinuousNumeric } from 'd3-scale';
+import { MutableRefObject, useEffect, useMemo, useState } from 'react';
+
+import { textDimentions } from '../utils';
+
+type Directions = 'horizontal' | 'vertical';
+
+export interface PrimaryLinearTicks {
+  label: string;
+  position: number;
+}
+
+interface Options {
+  tickFormat?: (d: number) => string;
+  scientificNotation?: boolean;
+  minSpace?: number;
+}
+
+const TEST_HEIGHT = '+1234567890';
+const MAX_ITERATION = 20;
+
+export function useLinearPrimaryTicks<
+  Scale extends ScaleContinuousNumeric<number, number>
+>(
+  scale: Scale,
+  direction: Directions,
+  ref: MutableRefObject<SVGGElement | null>,
+  options: Options = {},
+): PrimaryLinearTicks[] {
+  const [ticks, setTicks] = useState<number[]>([]);
+
+  const range = scale.range();
+  if (!range) throw new Error('Range needs to be specified');
+
+  const domain = scale.domain();
+  if (!domain) throw new Error('Domain needs to be specified');
+
+  const { scientificNotation = false, minSpace = 4 } = options;
+  const tickFormat: (d: number) => string = useMemo(
+    () =>
+      options.tickFormat || scientificNotation
+        ? (x) => x.toExponential(2)
+        : (x) => JSON.stringify(x),
+    [scientificNotation, options.tickFormat],
+  );
+
+  const axisLength = Math.abs(range[0] - range[1]);
+
+  // Calculates the tick number that fits in the given space
+  useEffect(() => {
+    if (ref.current) {
+      let tickNumber: number | undefined;
+      let repeat = true;
+      let ticks: number[] = [];
+
+      if (direction === 'horizontal') {
+        for (let count = 0; repeat && count < MAX_ITERATION; count++) {
```

Comment:
can we remove the MAX_ITERATION condition?are there any issues if you do not use it?

Reply:
I don't think that is a good practice to allow the possibility of an infinite loop, I already had the error of `Maximum update depth exceeded` in a previous case, so it is there for safety

Reply:
`Maximum update depth exceeded` happens with recursive calls, which I think you do not have. Running into an infinite loop is a bug in the code and should not be fixed with an arbitrary max iteration variable.